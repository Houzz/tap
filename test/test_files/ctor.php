<?php

class A {
    /**
     * @param int $a
     *
     */
    public function __construct($a) {
        echo $a + 1;
    }
}

$a = new A();           // error
$a = new A(1);

class B extends A {
    public function __construct() {
        parent::__construct();  // error
    }
}

class C extends A {
    public function __construct() {
        parent::__construct(1);
    }
}

$c = new C();
$c = new C(1);      // error

class D extends A {
    /**
     * @param string $x
     */
    function D($x) {
        parent::__construct(3);
    }
}

$d = new D();       // error
$d = new D(2);      // error
$d = new D("str");

