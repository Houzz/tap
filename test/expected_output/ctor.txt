13 E_FUNCTION_ARGUMENT
18 E_FUNCTION_ARGUMENT
29 E_FUNCTION_ARGUMENT
31 W_OLD_STYLE_CTOR
40 E_FUNCTION_ARGUMENT
41 E_FUNCTION_ARGUMENT
